contracts/NodeVersionBeacon.cdc

```diff
@@ -336,6 +350,17 @@ access(all) contract NodeVersionBeacon {
             // If we passed a boundary re-emit the VersionBeacon event
             NodeVersionBeacon.emitEventOnNextHeartbeat = true
         }
+
+        /// Emit a ProtocolStateVersionUpgrade event, if a pending upgrade is in storage.
+        access(self) fun emitProtocolStateVersionUpgradeEventIfNeeded() {
+            let pendingUpgrade = NodeVersionBeacon.popPendingProtocolStateVersionUpgrade()
+            if pendingUpgrade == nil {
+                return
+            }
+            emit NodeVersionBeacon.ProtocolStateVersionUpgrade(
+                newProtocolVersion: pendingUpgrade!.version,
+                activeView: pendingUpgrade!.activeView)
```

Comment:
Can use an if-let here instead of force-unwrapping:

```suggestion
            if let pendingUpgrade = NodeVersionBeacon.popPendingProtocolStateVersionUpgrade() {
                emit NodeVersionBeacon.ProtocolStateVersionUpgrade(
                    newProtocolVersion: pendingUpgrade.version,
                    activeView: pendingUpgrade.activeView
                )
            }
```